core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -433,7 +434,7 @@ public void run() {
               observer.onHeaders(headers);
             } catch (Throwable t) {
               cancel();
-              throw Throwables.propagate(t);
+              Throwables.propagate(t);
```

Comment:
The throw is to make it obvious that it is throwing. Also, it impacts Java's dead code and return detection. Having the throw is preferred.